### DISCUSSION TITLE
Don't cycle window focus

### DISCUSSION THREAD

**Discussion by gruvw** [qtile/qtile#3878] [Q&A] [answered]
Is there a way to not cycle through windows when using `lazy.layout.right()` (and the `left`, `down` and `up` versions) ?

Thank you.

**Comment by m-col** [comment, marked as answer]
Do you mean recursively wrapping around the list of windows? Some layouts have a parameter to configure this, for example `Columns` has `wrap_focus_columns` which is `True` by default - you'll need to see the docs for whichever layout(s) you're using.

**Reply by gruvw** [reply]
Yes thank you so much

**Reply by aloispichler** [reply]
I tried: `Key([mod], "h", lazy.layout.left( wrap_focus_columns= False))` but this does not seem to work as desired.
Where should I set this option? – Thanks

**Reply by m-col** [reply]
You need to pass it to the layout itself, e.g. `Columns(wrap_focus_columns=False)`, then your `lazy.layout.left()` will respect the option when you're on that layout.

**Reply by aloispichler** [reply]
Yes, I missed that. Thank you!

**Reply by clotodex** [reply]
@m-col is there a plan to make this an abstract function or config of ALL layouts? otherwise writing functions for or configuring multiple layouts is EXTREMELY layout dependent

**Reply by m-col** [reply]
@clotodex it would definitely be good for layouts that have this to use the same command name, so that the same binding applies to them all. Layouts for which it doesn't really apply (e.g. max, as there is only one window visible), firing the (undefined) command is essentially a no-op. But yep, anything to avoid super layout-dependent configuration is good :+1:
